economy/models.py

```diff
@@ -92,23 +51,156 @@ def add_funds(self, amount: int):
         self.balance += amount
         self.save()
 
+    # This intentionally allows setting a negative balance
     def remove_funds(self, amount: int):
         self.balance -= amount
         self.save()
 
 
 class SociProduct(models.Model):
     """
-    A product for sale at Soci
+    A product for sale at Soci.
+    Each product must have a unique SKU (stock keeping unit) identifier that enables us
+    to differentiate between different products in the Soci stock keeping system.
     """
-    sku_number = models.CharField(primary_key=True, unique=True, max_length=50, verbose_name='Product SKU number')
+    sku_number = models.CharField(unique=True, max_length=50, verbose_name='Product SKU number')
     name = models.CharField(max_length=50)
     price = models.IntegerField()
     description = models.TextField(blank=True, null=True, default=None, max_length=200)
-    icon = models.CharField(max_length=100)
+    icon = models.CharField(max_length=100, blank=True, null=True)
+    expiry_date = models.DateTimeField(blank=True, null=True, default=None)
 
     def __str__(self):
         return f"SociProduct {self.name} costing {self.price} kr"
 
     def __repr__(self):
         return f"SociProduct(name={self.name},price={self.price})"
+
+
+class ProductOrder(models.Model):
+    """
+    An order for a specific Soci product, and the order size.
+    If the product is direct charge, direct_charge_amount is used to specify the amount.
+    """
+    product = models.ForeignKey('SociProduct', on_delete=models.CASCADE)
+    order_size = models.IntegerField(default=1)
+    amount = models.IntegerField()
```

Comment:
Cannot `order_size` also indicate the amount on a direct charge?

Reply:
The intention was that since product prices might be updated in the future, in order to not mess up your transaction history, the charged amount would be stored in this field. It is not only used for direct charges.

Reply:
Fair enough!